stl/inc/memory

```diff
@@ -794,6 +794,24 @@ protected:
         }
     }
 
+    template <class _Ty2>
+    void _Weakly_construct_avoiding_expired_conversions_from(const _Ptr_base<_Ty2>& _Other) noexcept {
+        // implement weak_ptr's converting ctors
+        if (_Other._Rep) {
+            _Rep = _Other._Rep; // always share ownership
+            _Rep->_Incwref();

```

Comment:
Should we have an rvalue version of this function to avoid the extra increment and decrement of the weak reference count?

Reply:
Good idea! The original code predated rvalue references.